internal/core/app_status_report.go

```diff
@@ -333,6 +334,16 @@ func (op *statusReportOperation) Do(
 	}
 	realMsg.ResourcesHealth = resourcesHealth
 
+	// Add the deployment/release ID to the report. TODO: wire the resource ID through explicitly
+	switch op.Target.(type) {
```

Comment:
You want to use a type switch with an assignment here: https://golang.org/doc/effective_go#type_switch

That'll clean things up a lot for you.